include/rxmesh/matrix/dense_matrix.cuh

```diff
@@ -86,11 +96,15 @@ struct DenseMatrix
     T* col_data(const uint32_t ld_idx, locationT location = DEVICE) const
     {
         if ((location & HOST) == HOST) {
-            return m_h_val + ld_idx * lead_dim();
+            return m_h_val + ld_idx * (m_row_size + m_col_pad_idx);
         }
 
         if ((location & DEVICE) == DEVICE) {
-            return m_d_val + ld_idx * lead_dim();
+            return m_d_val + ld_idx * (m_row_size + m_col_pad_idx);
+        }
+
+        if ((location & m_allocated) == location) {
+            RXMESH_ERROR("Requested data not allocated on {}", location);
```

Comment:
`location` will print numeric value. I guess what we want here is `location_to_string(location)`